frontend/src/components/accountPage/UserList/UserList.tsx

```diff
@@ -0,0 +1,340 @@
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [searchText, setSearchText] = useState<string>('');
+
+  const closeModal = () => setshowUserListModal(false);
+
+  const onCsvUploaded = (fileInfo: any) => {
+    if (fileInfo.file.status === 'removed') {
+      setUsers([]);
+      return;
+    }
+
+    Papa.parse<any>(fileInfo.file, {
+      header: true,
+      skipEmptyLines: true,
+      complete: (result, _) => {
+        for (const line of result.data) {
+          if (
+            !line['NOME'] ||
+            !(line['COGNOME - (*) Inserito dal docente'] || line['COGNOME']) ||
+            !line['MATRICOLA'] ||
+            !line['EMAIL']
+          ) {
+            setFileError(
+              'Invalid file format, must contain <MATRICOLA, NOME, COGNOME (o COGNOME - (*) Inserito dal docente), EMAIL>'
+            );
+            return;
+          }
+        }
+        setUsers(() => {
+          const users = result.data.map((user: any, index: Number) => {
+            return {
+              name: user['NOME'] ?? '',
+              surname:
+                user['COGNOME - (*) Inserito dal docente'] ??
+                user['COGNOME'] ??
+                '',
+              userid: user['MATRICOLA'] ?? '',
+              email: user['EMAIL'] ?? '',
+              key: index,
+            };
+          });
+          props.onAddUser(users);
+          return users;
+        });
+
+        setFileError('');
+      },
+    });
+  };
+
+  return (
+    <>
+      <Table dataSource={users}>
+        <Column
+          title="User ID"
+          dataIndex="userid"
+          sorter={(a: User, b: User) => a.userid.localeCompare(b.userid)}
+          key="userid"
+          filterIcon={<SearchOutlined />}
+          onFilter={(value: any, record: any) =>
+            record['userid']
+              ? record['userid']
+                  .toString()
+                  .toLowerCase()
+                  .includes(value.toLowerCase())
+              : ''
+          }
+          filterDropdown={({
+            setSelectedKeys,
+            selectedKeys,
+            confirm,
+            clearFilters,
+          }) => (
+            <div style={{ padding: 8 }}>
+              <Input
+                placeholder="Search by Userid"
+                onChange={e => {
+                  setSelectedKeys(e.target.value ? [e.target.value] : []);
+                }}
+                value={selectedKeys[0]}
+                onPressEnter={() => {
+                  confirm();
+                  const currValue = selectedKeys[0]
+                    ? selectedKeys[0].toString()
+                    : '';
+                  setSearchText(currValue);
+                }}
+              />
+              <Space>
+                <Button
+                  onClick={() => {
+                    if (clearFilters) clearFilters();
+                    setSearchText('');
+                  }}
```

Comment:
This one can be extracted as well to avoid duplication.

Reply:
You could even make a sub component or just make a variable containing the button, in order to reuse it every time is needed.